### PR TITLE
test: add FixturesTester

### PR DESCRIPTION
**Pull request by armano2**
This PR changes way how alignment tests are generated.

`FixturesTester` is helper class which makes all what previously `createFixturePatternConfigFor`, `fixturesRequiringSourceTypeModule`, and those loops was doing before in way cleaner way.

### CODE REVIEW

File: tests/ast-alignment/fixtures-to-test.ts [hunk 1]
```diff
@@ -22,6 +22,65 @@ interface CreateFixturePatternConfig {
...
+    return this.fixtures
+      .map(fixture =>
+        glob
+          .sync(`${fixturesDirPath}/${fixture.pattern}`, {})
+          .map(filename => ({
+            filename,
+            ignoreSourceType: fixture.ignoreSourceType
+          }))
+      )
+      .reduce((acc, x) => acc.concat(x), []);
```
Comment (armano2) [inline tests/ast-alignment/fixtures-to-test.ts]: i was thinking about replacing this with `flatMap` but i will have to add `esnext` lib to tsconfig

**Comment by armano2** [comment]
github diff don't want to be nice and make diff readable...

**Comment by j-f1** [comment]
https://github.com/JamesHenry/typescript-estree/pull/67/files?w=1 might look a little better

**Comment by armano2** [comment]
@j-f1 what do you think about this anyway?

**Comment by JamesHenry** [comment]
I personally wouldn't have invested time in this, as it does not appear to add any new capabilities or fix any pain points vs the original code, but there is certainly nothing wrong with the changes you have made.

If you want to fix up the conflicts, I am happy to merge it in

**Comment by armano2** [comment]
i'm going to fix those merge conflicts

**Pull request merged**